src/murmur/main.cpp

```diff
@@ -200,6 +205,114 @@ void cleanup(int signum) {
 	exit(signum);
 }
 
+auto parseCLI(int argc, char **argv) {
+	struct {
+		bool quit = false;
+		boost::optional< std::string > ini_file;
+		boost::tuple< std::string, boost::optional< int > > supw_srv;
+		boost::optional< int > disable_su_srv;
+		bool verbose_logging = false;
+#ifdef QT_NO_DEBUG
+		bool detach = true;
+#else
+		boost::logic::tribool detach = boost::logic::indeterminate;
+#endif
+		bool wipe_ssl   = false;
+		bool wipe_logs  = false;
+		bool log_groups = false;
+		bool log_acls   = false;
+
+		bool print_authors            = false;
+		bool print_license            = false;
+		bool print_3rd_party_licenses = false;
+
+#ifdef Q_OS_UNIX
+		bool limits = false;
+		boost::optional< int > read_supw_srv;
+#endif
+	} options;
+
+	CLI::App app;
+	app.set_version_flag("-v,--version", "Mumble server version " + Version::getRelease().toStdString());
+
+	app.add_option_no_stream("-i,--ini", options.ini_file, "Specify ini file to use.")
+		->option_text("<inifile>")
+		->expected(1, 2)
+		->check(CLI::ExistingFile)
+		->group("Configuration");
+
+	app.add_option("-w,--supw", options.supw_srv, "Set password for 'SuperUser' account on server srv.")
+		->option_text("<pw> [srv]")
+		->allow_extra_args()
+		->expected(0, 1)
+		->group("Administration");
+
+#ifdef Q_OS_UNIX
+	app.add_option_no_stream("-r,--readsupw", options.read_supw_srv,
+							 "Reads password for server srv from standard input.")
+		->option_text("[srv]")
+		->default_val(1)
+		->expected(0, 1)
+		->group("Administration");
+
+	app.add_flag("-l,--limits", options.limits,
+				 "Tests and shows how many file descriptors and threads can be created.\n"
+				 "The purpose of this option is to test how many clients Murmur can handle.\n"
+				 "Murmur will exit after this test.")
+		->group("Testing");
+#endif
+
+	app.add_option_no_stream("-d,--disablesu", options.disable_su_srv,
+							 "Disable password for 'SuperUser' account on server srv.")
+		->option_text("[srv]")
+		->expected(0, 1)
+		->group("Administration");
+	app.add_flag("-s,--wipessl", options.wipe_ssl, "Remove SSL certificates from database.")->group("Administration");
+
+	app.add_flag("-b,--verbose", options.verbose_logging, "Use verbose logging (include debug-logs).")
```

Comment:
I think I'd rather make `--version` have no short option and make `-v` enable verbose mode. We're making a breaking change to the CLI anyway, so we might as well take the opportunity to streamline our options with what most other programs do as well.